### PR TITLE
Expose a minimal 2.0 API with the first summary endpoint.

### PR DESCRIPTION
**Pull request by rjnagal**
Right now, we just do raw container name. More types to follow.
We'll re-do the older endpoint in 2.0 before we publish it.

### CODE REVIEW

File: api/versions.go [hunk 1]
```diff
@@ -219,3 +221,43 @@ func (self *version1_2) HandleRequest(requestType string, request []string, m ma
...
+func (self *version2_0) Version() string {
+	return "v2.0"
+}
+
+func (self *version2_0) SupportedRequestTypes() []string {
+	return append(self.baseVersion.SupportedRequestTypes(), summaryApi)
+}
+
+func (self *version2_0) HandleRequest(requestType string, request []string, m manager.Manager, w http.ResponseWriter, r *http.Request) error {
+	switch {
```
Comment (vishh) [inline api/versions.go]: why not `switch requestType { case summaryApi: ... }`

**Comment by vishh** [comment]
LGTM.

File: api/versions.go (hunk 1 above)
Comment (vmarmol) [inline api/versions.go]: This is the style of the code above (maybe we should change all of it though).

File: api/versions.go [hunk 2]
```diff
@@ -219,3 +221,43 @@ func (self *version1_2) HandleRequest(requestType string, request []string, m ma
 		return self.baseVersion.HandleRequest(requestType, request, m, w, r)
 	}
 }
+
+// v2.0 builds on v1.2
+type version2_0 struct {
+	baseVersion *version1_2
+}
+
+func newVersion2_0(v *version1_2) *version2_0 {
```
Comment (vmarmol) [inline api/versions.go]: I think v2.0 shouldn't be based on v1.X, it'll probably be separate completely.

Maybe we keep this for now since it is not complete though. Although having an incomplete WIP API will probably discourage others from using it until it is ready :)

File: api/versions.go [hunk 3]
```diff
@@ -219,3 +221,43 @@ func (self *version1_2) HandleRequest(requestType string, request []string, m ma
...
+	case requestType == summaryApi:
+		containerName := getContainerName(request)
+		glog.V(2).Infof("Api - Summary(%v)", containerName)
+
+		stats, err := m.GetContainerDerivedStats(containerName)
+		if err != nil {
+			return err
+		}
+
+		err = writeResult(stats, w)
```
Comment (vmarmol) [inline api/versions.go]: nit: return writeResult()

File: manager/manager.go [hunk 4]
```diff
@@ -364,6 +367,20 @@ func (self *manager) containerDataSliceToContainerInfoSlice(containers []*contai
 	return output, nil
 }
 
+func (self *manager) GetContainerDerivedStats(containerName string) (info.DerivedStats, error) {
+	var ok bool
+	var cont *containerData
+	func() {
```
Comment (vmarmol) [inline manager/manager.go]: unrelated and not for this PR: we do this everywhere, we should get a helper...

**Comment by vmarmol** [comment]
LGTM, just some discussion points and nits

File: manager/manager.go (hunk 4 above)
Comment (vishh) [inline manager/manager.go]: +1. I was thinking the same thing.

File: api/versions.go (hunk 1 above)
Comment (rjnagal) [inline api/versions.go]: Done here and everywhere else.

File: api/versions.go (hunk 2 above)
Comment (rjnagal) [inline api/versions.go]: I want to keep it based off 1.2 and replace whatever APIs we need to change. My guess is that some of the APIs will stay the same. If we rip out everything, we can remove the base dependency. 

I am fine going the other way and adding the base versions once we know that some of the endpoints will be replicated.

File: api/versions.go (hunk 3 above)
Comment (rjnagal) [inline api/versions.go]: done

File: manager/manager.go (hunk 4 above)
Comment (rjnagal) [inline manager/manager.go]: I was too lazy this time around. Will do a separate PR :)

**Comment by vmarmol** [comment]
LGTM
